frontend/npm.go

```diff
@@ -2,54 +2,53 @@ package frontend
 	"context"
-	"encoding/json"
 	"fmt"
 	"log/slog"
-	"strings"
 
 	"dagger.io/dagger"
 	"github.com/grafana/grafana-build/containers"
-	"github.com/grafana/grafana-build/daggerutil"
 )
 
 // NPMPackages versions and packs the npm packages into tarballs into `npm-packages` directory.
 // It then returns the npm-packages directory as a dagger.Directory.
 func NPMPackages(ctx context.Context, builder *dagger.Container, d *dagger.Client, log *slog.Logger, src *dagger.Directory, ersion string) (*dagger.Directory, error) {
 	// Check if the version of Grafana uses lerna or nx to manage package versioning.
-	hasLerna := daggerutil.FileExists(ctx, src, "./lerna.json")
-	hasNx := daggerutil.FileExists(ctx, src, "./nx.json")
-
-	if hasLerna {
-		return builder.WithExec([]string{"mkdir", "npm-packages"}).
-			WithExec([]string{"yarn", "packages:build"}).
-			WithExec([]string{"yarn", "lerna", "version", ersion, "--exact", "--no-git-tag-version", "--no-push", "--force-publish", "-y"}).
-			WithExec([]string{"yarn", "lerna", "exec", "--no-private", "--", "yarn", "pack", "--out", fmt.Sprintf("/src/npm-packages/%%s-%v.tgz", "v"+ersion)}).
-			Directory("./npm-packages"), nil
-	}
-
-	if hasNx {
-		packages, err := containers.GetJSONValue(ctx, d, src, "nx.json", "release.groups.grafanaPackages.projects")
-		if err != nil {
-			return nil, err
-		}
-
-		var packagesToPack []string
-		err = json.Unmarshal(([]byte)(packages), &packagesToPack)
-		if err != nil {
-			return nil, err
-		}
-
-		grafanaPackages := strings.Join(packagesToPack, ",")
-
-		return builder.WithExec([]string{"mkdir", "npm-packages"}).
-			WithExec([]string{"yarn", "packages:build"}).
-			WithExec([]string{"yarn", "nx", "release", "version", ersion, "--no-git-commit", "--no-git-tag", "--no-stage-changes", "--group", "grafanaPackages"}).
-			WithExec([]string{"yarn", "nx", "exec", fmt.Sprintf(`--projects="%s"`, grafanaPackages), "--", "yarn", "pack", "--out", fmt.Sprintf("/src/npm-packages/%%s-%v.tgz", "v"+ersion)}).
-			Directory("./npm-packages"), nil
-	}
-
-	log.Error("No lerna.json or nx.json found in source directory")
-	return nil, fmt.Errorf("no lerna.json or nx.json found in source directory")
+	var (
+		out = fmt.Sprintf("/src/npm-packages/%%s-%v.tgz", "v"+ersion)
+
+		lernaBuild = fmt.Sprintf("yarn lerna version %s --exact --no-git-tag-version --no-push --force-publish -y", ersion)
+		lernaPack  = fmt.Sprintf("yarn lerna exec --no-private -- yarn pack --out %s", out)
+
+		nxBuild = fmt.Sprintf("yarn nx release version %s --no-git-commit --no-git-tag --no-stage-changes --group grafanaPackages", ersion)
+		nxPack  = fmt.Sprintf("yarn nx exec --projects=$(cat nx.json | jq -r '.relase.groups.grafanaPackages.projects | join(\",\")') -- yarn pack --out %s", out)
+	)
+
+	return builder.WithExec([]string{"mkdir", "npm-packages"}).
+		WithExec([]string{"yarn", "packages:build"}).
+		WithExec([]string{"/bin/sh", "-c", fmt.Sprintf("if [ -f .nx ]; then %s; else %s; fi", nxBuild, lernaBuild)}).
```

Comment:
This checks if a file called `.nx` exists? There's a directory called `.nx` but that exists before and after the removal of lerna. It also doesn't exist in CI. I think we should check if the `lerna.json` file exists or not for backwards compatibility as it is [deleted in the associated branch](https://github.com/grafana/grafana/pull/85403/files#diff-2d72bdead8afa0798d18995311992d684348a694c2d5e214e8e4d2b6153e4821).

Reply:
So if the lerna.json file is there, then use the lerna command, otherwise use the nx command?

Reply:
Exactly! Thanks for updating it. I'll take it for a spin against main and the associated grafana dev branch shortly.